tooling/image-sync/internal/sync.go

```diff
@@ -103,19 +133,46 @@ func DoSync(cfg *SyncConfig) error {
 	Log().Infow("Syncing images", "images", cfg.Repositories, "numberoftags", cfg.NumberOfTags)
 	ctx := context.Background()
 
-	quaySecret, err := readQuaySecret(cfg.QuaySecretFile)
-	if err != nil {
-		return fmt.Errorf("error reading secret file: %w %s", err, cfg.QuaySecretFile)
+	srcRegistries := make(map[string]Registry)
+	var err error
+
+	for _, secret := range cfg.Secrets {
+		if secret.Registry == "quay.io" {
+			quaySecret, err := readBearerSecret(secret.SecretFile)
+			if err != nil {
+				return fmt.Errorf("error reading secret file: %w %s", err, secret.SecretFile)
+			}
+			qr := NewQuayRegistry(cfg, quaySecret.BearerToken)
+			srcRegistries[secret.Registry] = qr
+		} else {
+			if strings.HasSuffix(secret.Registry, "azurecr.io") {
```

Comment:
depending on the involved cloud, the ACR endpoint suffix is slightly different

```
az cloud list | jq '.[] | .name, .suffixes.acrLoginServerEndpoint'
"AzureCloud"
".azurecr.io"
"AzureChinaCloud"
".azurecr.cn"
"AzureUSGovernment"
".azurecr.us"
"AzureGermanCloud"
null
```

Reply:
I'll try to check if i can retrieve that list from the SDK